pkg/endpoint/bpf.go

```diff
@@ -62,15 +62,6 @@ func (e *Endpoint) writeL4Map(fw *bufio.Writer, owner Owner, m policy.L4PolicyMa
 		dport := byteorder.HostToNetwork(uint16(l4.Port))
 
 		redirect := uint16(l4.L7RedirectPort)
-		if l4.IsRedirect() && redirect == 0 {
-			redirect, err = e.addRedirect(owner, &l4)
```

Comment:
You can't remove the `addRedirect` from here because this is the function that allocate the redirect port. The redirect is also used on line 66.

Reply:
What's the dependency? It's still being allocated under the same endpoint.Mutex, just not so deep in the stack.